src/braket/pennylane_plugin/braket_device.py

```diff
@@ -1012,6 +1023,30 @@ def __init__(
         device = LocalSimulator(backend)
         super().__init__(wires, device, shots=shots, **run_kwargs)
 
+    def _run_task_batch(self, batch_circuits, batch_shots: int, inputs):
+        task_batch = self._device.run_batch(
+            batch_circuits,
+            shots=batch_shots,
+            max_parallel=self._max_parallel,
+            inputs=inputs,
+            **self._run_kwargs,
+        )
+
+        # Should not need try-except here as this is a local sim.
+        braket_results_batch = task_batch.results()
+
+        # Update the tracker
+        if self.tracker.active:
+            for task in task_batch.tasks:
+                tracking_data = self._tracking_data(task)
+                self.tracker.update(**tracking_data)
+            total_executions = len(task_batch.tasks) - len(task_batch.unsuccessful)
+            total_shots = total_executions * batch_shots
+            self.tracker.update(batches=1, executions=total_executions, shots=total_shots)
+            self.tracker.record()
```

Comment:
Why include this for the local simulator? As well, this is the same code as in `BraketAwsQubitDevice`, so extract to a common method if you do need it.

Reply:
A user might want to use these data to extrapolate what it would have cost to run this PL workflow on a QPU